### DISCUSSION TITLE
How to connect the Zigbee Tuya Siren TS0224 to HA  with Z2M

### DISCUSSION THREAD

**Discussion by johanstaal** [Koenkk/zigbee2mqtt#20920] [Devices]
I am setting up an alarm system in Home Assistant based on Alarmo. Everything seems to work fine, but I am not able to get my Zigbee siren (Tuya TS0224, Meian brand) working. 

I have integrated the siren in Zigbee2MQTT where it shows up as expected:

![image](https://github.com/Koenkk/zigbee2mqtt/assets/45044370/4769175c-83c2-45e4-a0fa-a7e081e060d3)

When I select Exposes I get this:

![image (1)](https://github.com/Koenkk/zigbee2mqtt/assets/45044370/2f3abe81-be21-400e-a7fc-ac1bfc99e9a3)

When I fill in the desired values for mode, volume etc. and press Apply the siren starts screaming, so there is communication with the siren. However, I can only select modes “Burglar” and “Stop”, and the volume is always the same level, whatever level I choose. Same for strobe settings, the strobe does not work.That is not a problem for me, but it is not what I expect. In addition, I get an error message, when I fill in the volume level in the lower part of the form.The error message tells this:

![905edbe4f3c21f16e9af0b32b207980931f68370](https://github.com/Koenkk/zigbee2mqtt/assets/45044370/e0e3cfc0-37b0-4ac8-98af-babab75c0369)

I was able to make two scenes in Z2M, one for "burglar"and one for “stop”, but these scenes do not show up in the scenes list in HA.
Further I tried to use services turn-on and turn_off for the siren but that did not work either.

Who can help me get this siren working with Alarmo?

**Comment by DaviRiver** [comment]
I have the same isue, please, any update??
Thaks!!

**Reply by johanstaal** [reply]
No, unfortunately no reaction yet.

**Reply by greboreda** [reply]
Same issue here. Any updates would be greatly appreciated :)

thxxx

**Comment by Millio345** [comment]
Same issue, will look into this further soon, meanwhile at least the original integration issue contains a lot of additional information:
https://github.com/Koenkk/zigbee2mqtt/issues/16060

**Comment by danielrosehill** [comment]
I'm able to adjust the parameters but the scenes that I create from within Z2M don't work in HA. Anyone experiencing the same thing?

**Comment by chpmc** [comment]
Same problems here. Adjusting the volume level and almost all other settings has no effect. Neither in the Z2M device page nor via scripts.

**Comment by manny199** [comment]
The problem persist. Not working sound on home assistant.
Is there a solution?
